app/Http/Controllers/SettingsController.php

```diff
@@ -341,6 +341,13 @@ public function postSettings(Request $request)
         }
 
         $setting->full_multiple_companies_support = $request->input('full_multiple_companies_support', '0');
+        $setting->scope_locations_fmcs = $request->input('scope_locations_fmcs', '0');
+
+        // Backward compatibility for locations makes no sense without FullMultipleCompanySupport
+        if (!$setting->full_multiple_companies_support) {
+            $setting->scope_locations_fmcs = '0';
+        }
```

Comment:
I think this can be handled in the view with some javascript. If this is a problem I can think about it some more, but my experience with this is quite limited.